spec.bs

```diff
@@ -1839,6 +1936,34 @@ Issue(WICG/turtledove#522): Move from "`*`" to "`self`".
 
 # Structures # {#structures}
 
+
+<xmp class="idl">
+
+dictionary GenerateBidInterestGroup {
+  required USVString owner;
+  required USVString name;
+
+  double priority = 0.0;
+  boolean enableBiddingSignalsPrioritization = false;
+
+  DOMString executionMode = "compatibility";
+  USVString biddingLogicURL;
+  USVString biddingWasmHelperURL;
+  USVString updateURL;
+  USVString trustedBiddingSignalsURL;
+  sequence<USVString> trustedBiddingSignalsKeys;
+  any userBiddingSignals;

```

Comment:
I think this is a DOMString or USVString, and similarly for the metadata field in the nested `ads` and `adComponents` types.

Reply:
It was my mistake. They are not strings, but anything that can be converted from/to JSON, so the interest group IDL set them as `any`.
When passed to generateBid(), they will be the result of parsing stored JSON, which can be an object, an integer, etc. 
https://crbug.com/1238147
https://github.com/WICG/turtledove/blob/main/FLEDGE.md#32-on-device-bidding